rust/src/web/graphql/feetransfers/mod.rs

```diff
@@ -0,0 +1,237 @@
+use super::blocks::{Block, BlockWithCanonicity};
+use crate::{
+    block::{precomputed::PrecomputedBlock, store::BlockStore, BlockHash},
+    canonicity::{store::CanonicityStore, Canonicity},
+    command::{internal::InternalCommandWithData, store::CommandStore},
+    constants::MAINNET_GENESIS_HASH,
+    store::IndexerStore,
+};
+use async_graphql::{Context, Enum, InputObject, Object, Result, SimpleObject};
+use chrono::{DateTime, SecondsFormat};
+use std::sync::Arc;
+
+#[derive(SimpleObject)]
+pub struct Feetransfer {
+    pub state_hash: String,
+    pub fee: u64,
+    pub recipient: String,
+    #[graphql(name = "type")]
+    pub feetransfer_kind: String,
+    pub block_height: u32,
+    pub date_time: String,
+}
+
+pub struct FeetransferWithMeta {
+    /// Value canonicity
+    pub canonical: bool,
+    /// Value optional block
+    pub block: Option<PrecomputedBlock>,
+    /// Value feetranser
+    pub feetransfer: Feetransfer,
+}
+
+#[Object]
+impl FeetransferWithMeta {
+    async fn canonical(&self) -> bool {
+        self.canonical
+    }
+
+    #[graphql(flatten)]
+    async fn feetransfer(&self) -> &Feetransfer {
+        &self.feetransfer
+    }
+
+    async fn block_state_hash(&self) -> Option<BlockWithCanonicity> {
+        self.block.clone().map(|block| BlockWithCanonicity {
+            block: Block::from(block),
+            canonical: self.canonical,
+        })
+    }
+}
+
+#[derive(InputObject, Clone)]
+pub struct FeetransferQueryInput {
+    state_hash: Option<String>,
+    canonical: Option<bool>,
+}
+
+#[derive(Enum, Copy, Clone, Eq, PartialEq)]
+pub enum FeetransferSortByInput {
+    #[graphql(name = "BLOCKHEIGHT_ASC")]
+    BlockHeightAsc,
+    #[graphql(name = "BLOCKHEIGHT_DESC")]
+    BlockHeightDesc,
+}
+
+#[derive(Default)]
+pub struct FeetransferQueryRoot;
+
+/// convert epoch millis to an ISO 8601 formatted date
+fn millis_to_date_string(millis: i64) -> String {
+    let date_time = DateTime::from_timestamp_millis(millis).unwrap();
+    date_time.to_rfc3339_opts(SecondsFormat::Millis, true)
+}
+
+#[Object]
+impl FeetransferQueryRoot {
+    async fn feetransfers<'ctx>(
+        &self,
+        ctx: &Context<'ctx>,
+        query: Option<FeetransferQueryInput>,
+        sort_by: Option<FeetransferSortByInput>,
+        limit: Option<usize>,
+    ) -> Result<Option<Vec<FeetransferWithMeta>>> {
+        let db = ctx
+            .data::<Arc<IndexerStore>>()
+            .expect("db to be in context");
```

Comment:
```suggestion
        let db = db(ctx);
```

[Function details](https://github.com/Granola-Team/mina-indexer/blob/main/rust/src/web/graphql/mod.rs#L37)